blogposts/inside-rust/01-announcement.md

```diff
@@ -0,0 +1,162 @@
+
+The FFI-unwind project group, announced in [this RFC][rfc-announcement], is
+working to extend the language to support unwinding that crosses FFI
+boundaries.
+
+We have reached our first technical decision point, on a question we have been
+discussing internally for quite a while. This blog post lays out the arguments
+on each side of the issue and invites the Rust community to join us at the
+[upcoming meeting](meeting-link) to help finalize our decision, which will be
+formalized and published as our first language-change RFC. This RFC will
+propose an "MVP" specification for well-defined cross-language unwinding.
+
+The meeting will be on Monday the 24th at 17:00 UTC, via Zoom:
+
+> Meeting ID: 768 231 760
+> https://mozilla.zoom.us/j/768231760
+
+## Background: what is unwinding?
+
+Exceptions are a familiar control flow mechanism in many programming languages.
+They are particularly commonplace in managed languages such as Java, but they
+are also part of the C++ language, which introduced them to the world of
+unmanaged systems programming.
+
+When an exception is thrown, the runtime _unwinds_ the stack, essentially
+traversing it backwards and calling clean-up or error-recovery code such as
+destructors or `catch` blocks.
+
+Compilers may implement their own unwinding mechanisms, but in native code such
+as Rust binaries, the mechanism is more commonly provided by the platform ABI.
+
+It is well known that Rust does not have exceptions as such. But Rust _does_
+support unwinding! There are two scenarios that will cause unwinding to occur:
+
+* By default, Rust's `panic!()` unwinds the stack.
+* Using FFI, Rust can call functions in other languages (such as C++) that can
+  unwind the stack.
+  * There are some special cases where C libraries can actually cause
+    unwinding.  For instance, on Microsoft platforms, `longjmp` is implemented
+    as "forced unwinding" (see below)
+
+Currently, when foreign (non-Rust) code invokes Rust code, the behavior of a
+`panic!()` unwind that "escapes" from Rust is explicitly undefined. Similarly,
+when Rust calls a foreign function that unwinds, the behavior once the unwind
+operation encounters Rust frames is undefined. The primary reason for this is
+to ensure that Rust implementations may use their own unwinding mechanism,
+which may not be compatible with the platform-provided "native" unwinding
+mechanism. Currently, however, `rustc` uses the native mechanism, and there are
+no plans to change this.
+
+### Forced unwinding
+
+Platform ABIs can define a special kind of unwinding called "forced unwinding."
+This type of unwinding is never supposed to be intercepted by the language
+runtime; so, for instance, C++ destructors should not be invoked.
+
+There are two common types of forced unwinding:
+
+* On Windows platforms, `longjmp` is implemented as a forced unwind.
+* On UNIX-like platforms, canceling a `phtread` results in a forced unwind.
```

Comment:
Typo: pthread. Also I would use `pthread_exit` as an example since that is likely to be more familiar to readers.

Reply:
Should I state the function name, or just say "exiting or canceling a `pthread`? (Does it matter?)

Reply:
I would suggest explicitly naming `pthread_exit`.